kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/Dek.java

```diff
@@ -203,11 +195,12 @@ private static long commenceDestroyBoth(long combined) {
         this.atomicKey = new AtomicReference<>(key);
         this.cipherSpec = cipherSpec;
         this.remainingEncryptions = new AtomicLong(maxEncryptions);
-        this.outstandingCryptors = new AtomicLong(START);
+        // If no encryptions then make the Dek already destroyed for encrypt.
+        this.outstandingCryptors = new AtomicLong(maxEncryptions == 0 ? combine(-1, 1) : START);
```

Comment:
Ok, I admit to being late to this bit twiddling party but... 

I feel like we should have our own type which wraps the Atomic long and exposes getters and setters for encryptors and decryptors. In fact we already have most of the API sketched out/ implemented inline e.g `encryptorCount` `decryptorCount`. 

This doesn't need to happen(if we agree that we should do it) before this PR merges!

Reply:
Rob's done that in https://github.com/kroxylicious/kroxylicious/pull/959/, so let's discuss this point there.